Field2d widget causes exception every frame

**Describe the bug**  
Log file gets spammed with error: 
```
Mar XX, 2024 XX:XX:XX XM edu.wpi.first.shuffleboard.plugin.networktables.sources.CompositeNetworkTableSource lambda$new$0
WARNING: Incomplete data for type Field2d
edu.wpi.first.shuffleboard.api.data.IncompleteDataException: Incomplete data in map: {.name=Odom, .type=Field2d, .controllable=true, Robot=[D@35f5daf9,  Localized=x: 1.4519350881997504, y: -31.02742566505306}
	at edu.wpi.first.shuffleboard.api.data.ComplexDataType.fromMap(ComplexDataType.java:33)
	at edu.wpi.first.shuffleboard.plugin.networktables.sources.CompositeNetworkTableSource.lambda$new$0(CompositeNetworkTableSource.java:58)
	at edu.wpi.first.shuffleboard.plugin.networktables.sources.NetworkTableSource.lambda$setTableListener$0(NetworkTableSource.java:81)
	at edu.wpi.first.shuffleboard.api.util.FxUtils.lambda$runOnFxThread$0(FxUtils.java:64)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class [D (java.lang.String and [D are in module java.base of loader 'bootstrap')
	at edu.wpi.first.shuffleboard.plugin.base.data.FieldData.<init>(FieldData.java:73)
	at edu.wpi.first.shuffleboard.api.data.ComplexDataType.fromMap(ComplexDataType.java:31)
	... 10 more
```

**To Reproduce**  
1. Send ANY Field2d widget using SmartDashboard from robot code
2. Exception gets thrown every frame

**Expected behavior**  
No exception gets thrown, and robot pose is correctly displayed

**Actual behavior**
Field2d widget causes exception (every frame?), and robot pose is fixed at 0, 0

**Screenshots**  
Also causes MASSIVE log file spam
10 GB+ log files:
![image](https://github.com/wpilibsuite/shuffleboard/assets/25181276/4fd7fb25-11a5-4050-9ef6-03cb2d3929a2)
Smaller log file:
[shuffleboard.2024-03-07-11.44.02.log](https://github.com/wpilibsuite/shuffleboard/files/14614965/shuffleboard.2024-03-07-11.44.02.log)


**Desktop:**
 - OS: Windows
 - Java version: 17.0.6
 - Version: 2024.2.1
![image](https://github.com/wpilibsuite/shuffleboard/assets/25181276/c1985e69-3207-4760-997d-7cd44155cb57)

--- a/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/FieldData.java
+++ b/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/FieldData.java
@@ -6,6 +6,7 @@ import edu.wpi.first.shuffleboard.api.data.ComplexData;
 import java.nio.ByteBuffer;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.logging.Logger;
 
 public class FieldData extends ComplexData<FieldData> {
   private final SimplePose2d robot;
@@ -60,8 +61,13 @@ public class FieldData extends ComplexData<FieldData> {
       }
 
       double[] doubles;
-      if (entry.getValue() instanceof byte[]) {
-        byte[] data = (byte[]) entry.getValue();
+      Object value = entry.getValue();
+      if (value instanceof double[]) {
+        // Intended flow
+        doubles = (double[]) value;
+      } else if (entry.getValue() instanceof byte[]) {
+        // double[] encoded as byte[] is supported too (for some reason)
+        byte[] data = (byte[]) value;
         doubles = new double[data.length / Double.BYTES];
         for (int i = 0; i < doubles.length; i++) {
           doubles[i] = ByteBuffer.allocate(Double.BYTES)
@@ -70,7 +76,13 @@ public class FieldData extends ComplexData<FieldData> {
                   .getDouble();
         }
       } else {
-        doubles = (double[]) map.get(key);
+        // Some other type (might be an externally-published topic under the same table)
+        // Warn and drop, but don't raise an exception
+        doubles = new double[0];
+        Logger.getLogger(FieldData.class.getName())
+                .warning(() -> String.format(
+                        "Failed to parse entry %s of type %s: %s",
+                        key, value.getClass(), value));
       }
 
       SimplePose2d[] poses = new SimplePose2d[doubles.length / 3];

--- a/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/FieldData.java
+++ b/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/FieldData.java
@@ -65,7 +65,7 @@ public class FieldData extends ComplexData<FieldData> {
       if (value instanceof double[]) {
         // Intended flow
         doubles = (double[]) value;
-      } else if (entry.getValue() instanceof byte[]) {
+      } else if (value instanceof byte[]) {
         // double[] encoded as byte[] is supported too (for some reason)
         byte[] data = (byte[]) value;
         doubles = new double[data.length / Double.BYTES];

--- a/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/FieldData.java
+++ b/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/FieldData.java
@@ -104,6 +104,8 @@ public class FieldData extends ComplexData<FieldData> {
 
   @Override
   public Map<String, Object> asMap() {
-    return ImmutableMap.of("Robot", robot);
+    HashMap<String, SimplePose2d[]> tmp = new HashMap<>(getObjects());
+    tmp.put("Robot", new SimplePose2d[]{getRobot()});
+    return ImmutableMap.copyOf(tmp);
   }
 }

